### PR TITLE
Make `ed25519`, `blake2b` and `curl-p` features `no_std` compatible.

### PR DESCRIPTION
**Pull request by pvdrz**
Blocked by https://github.com/iotaledger/crypto.rs/pull/109

### CODE REVIEW

**Review by Wollac — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -81,7 +81,7 @@ blake2 = { version = "0.9", optional = true, default-features = false }
 byteorder = { version = "1.4", optional = true, default-features = false }
 chacha20poly1305 = { version = "0.8", optional = true }
 digest = { version = "0.9", optional = true, default-features = false }
-ed25519-zebra = { version = "2.2", optional = true, default-features = false }
+ed25519-zebra = { version = "3.0", git = "https://github.com/ZcashFoundation/ed25519-zebra", rev = "15e0286", optional = true, default-features = false }
```
Comment (semenov-vladyslav) [inline Cargo.toml]: Why not `tag = "3.0.0"`?

File: src/hashes/ternary/curl_p/regular/transform.rs [hunk 2]
```diff
@@ -23,22 +19,32 @@ struct StateRotation {
...
-        state_rotations
-    };
-}
+    while i < NUM_ROUNDS {
+        state_rotations[i].offset = rotation / HASH_LENGTH;
+        state_rotations[i].shift = {
+            let shift = rotation % HASH_LENGTH;
+
+            if shift > 255 {
+                panic!("shift is too large");
```
Comment (semenov-vladyslav) [inline src/hashes/ternary/curl_p/regular/transform.rs:35]: `HASH_LENGTH = 243` , this branch is impossible.

**Review by semenov-vladyslav — CHANGES_REQUESTED** [review]
Missing .changes file. Everything else looks OK.

File: Cargo.toml (hunk 1 above)
Comment (semenov-vladyslav) [reply Cargo.toml]: This is addressed in #109.

File: src/hashes/ternary/curl_p/regular/transform.rs (hunk 2 above)
Comment (pvdrz) [reply src/hashes/ternary/curl_p/regular/transform.rs:35]: Yeah I know, that's why this is done during compilation so if someone changes the code logic they actually notice that they messed up.

**Review by Wollac — APPROVED** [review]

**Review by semenov-vladyslav — APPROVED** [review]
Looks good

**Pull request merged**